src/pybamm/dispatch/entry_points.py

```
@@ -0,0 +1,168 @@
+
+    def _load_entry_point(self, key) -> Callable:
+        """Check that ``key`` is a registered ``parameter_sets`` or ``models` ,
+        and return the entry point for the parameter set/model, loading it needed."""
+        if key not in self._all_entries:
+            raise KeyError(f"Unknown parameter set or model: {key}")
+        param_set = self._all_entries[key]
+        try:
+            param_set = self._all_entries[key] = param_set.load()
+        except AttributeError:
+            pass
+        return param_set
+
+    def __iter__(self):
+        return self._all_entries.__iter__()
+
+    def __len__(self) -> int:
+        return len(self._all_entries)
+
+    def get_docstring(self, key):
+        """Return the docstring for the ``key`` parameter set or model"""
+        return textwrap.dedent(self._load_entry_point(key).__doc__)
+
+    def __getattribute__(self, name):
+        try:
+            # For backwards compatibility, parameter sets that used to be defined in
+            # this file now return the name as a string, which will load the same
+            # parameter set as before when passed to `ParameterValues`
+            # Bypass the overloaded __getitem__ and __iter__ to avoid recursion
+            _all_entries = super().__getattribute__("_all_entries")
+            group = super().__getattribute__("group")
+
+            # Only show deprecation warning for parameter sets, not models
+            if (
+                name in _all_entries and group == "pybamm_parameter_sets"
+            ):  # pragma: no cover
+                msg = (
+                    f"Parameter sets should be called directly by their name ({name}), "
+                    f"instead of via pybamm.parameter_sets (pybamm.parameter_sets.{name})."
+                )
+                warnings.warn(msg, DeprecationWarning, stacklevel=2)
+                return name
+        except AttributeError:
+            pass  # Handle the attribute error normally
```

This looks more like it ignores the error than handles it normally